darts/models/filtering_model.py

```diff
@@ -0,0 +1,108 @@
+"""
+Filtering Model Base Class
+------------------------------
+
+A filtering model captures the measured values of a time series as a function of the past as follows:
+
+.. math:: y_{t+1} = f(y_{t+1}, y_t, y_{t-1}, ..., y_1),
+
+where :math:`y_t` represents the time series' filtered value(s) at time :math:`t`.
+
+The main functions are `fit()` and `predict()`. `fit()` learns the function `f()`, over the history of
+one or several time series. The function `predict()` applies `f()` on one or several time series in order
+to obtain more accurate predictions of the measured values(s)
+"""
+
+from typing import Optional
+from itertools import product
+from abc import ABC, abstractmethod
+from inspect import signature
+import numpy as np
+import pandas as pd
+
+from ..timeseries import TimeSeries
+from ..logging import get_logger, raise_log, raise_if_not
+
+logger = get_logger(__name__)
+
+
+class FilteringModel(ABC):
+
+    """ The base class for filtering models. It defines the *minimal* behavior that all filtering models have to support.
+        The signatures in this base class are for "local" models handling only one series and no covariates.
+        Sub-classes can handle more complex cases.
+    """
+    @abstractmethod
+    def __init__(self):
+        # The series used for training the model through the `fit()` function.
+        # This is only used if the model has been fit on one time series.
+        self.training_series: Optional[TimeSeries] = None
+
+        # state; whether the model has been fit (on a single time series)
+        self._fit_called = False
+
+    @abstractmethod
+    def fit(self, series: TimeSeries) -> None:
```

Comment:
Is there a need for a `fit()` method for these models?

Reply:
just to be able to pass the data and not consume CPU time yet?

Reply:
I don't think that's really something that would be needed in practice. I really think all we need is a `predict(series: TimeSeries)` function. Adding a `fit()` that does nothing complexifies the API (and `fit()` would be a poor name for a function that does nothing except storing the data).

Reply:
I am keeping just the "filter" function, to keep the naming in sync with model name. fit function remvoed and kf re-initialized.